### ISSUE TITLE
[core] Exception in thread "main" java.util.ConcurrentModificationException

### ISSUE THREAD

**Issue by flankerhqd** [skylot/jadx#1488]
Hi:

A ConcurrentModificationException occurred when using jadx api to visit classes and method.

The stacktrace is as follows:

```
Exception in thread "main" java.util.ConcurrentModificationException
	at java.base/java.util.ArrayList$Itr.checkForComodification(ArrayList.java:1042)
	at java.base/java.util.ArrayList$Itr.next(ArrayList.java:996)
	at jadx.api.JavaClass.loadLists(JavaClass.java:113)
	at jadx.api.JavaClass.getMethods(JavaClass.java:279)
	at Main.visitClass(Main.java:50)
	at Main.main(Main.java:44)
```

Reproduce code:

```java
    JadxArgs jadxArgs = new JadxArgs();
    jadxArgs.setInputFile(new File("sample.apk"));
    jadxArgs.setOutDir(new File("apkDecompile/"));
    JadxDecompiler jadx = new JadxDecompiler(jadxArgs);
    jadx.load();

    for (JavaClass cls : jadx.getClasses()) {
      cls.getMethods();
    }
```

The exception seems to be stablely reproduceable on  1.3.5.

The sample apk can be downloaded at https://transfer.sh/G6fiXN/sample.apk

**Comment by flankerhqd** [comment]
Hello:

Nope, this is the exact code. (The machine has 8 cores)

Please try this link https://blog.flanker017.me/wp-content/uploads/2022/05/sample.tar.gz and rename it to apk.

Thanks!

**Comment by skylot** [comment]
@flankerhqd I successfully reproduce this issue using jadx 1.3.5. But using the latest commit from master, this issue not appear anymore. I will release new jadx version tomorrow, so you will be able to check and confirm the fix.

**Comment by flankerhqd** [comment]
Great, thanks

**Comment by flankerhqd** [comment]
Verified to be fixed in maven central 1.4.0.
